### PR TITLE
Feat: update marketplaces

### LINKED ISSUE #366: Add SG to advertising
Seems to be added.

Need to add:

 * Time zone
 * Bids
 * Add advertising to `sg.ts`

https://advertising.amazon.com/API/docs/en-us/get-started/how-to-use-api

image

### LINKED ISSUE #399: Change EU Seller Central URIs
Docs: https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#seller-central-urls

Use `https://sellercentral-europe.amazon.com` where specified.

### LINKED ISSUE #400: Add Vendor Central URL
Docs: https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#vendor-central-urls

Add after `sellerCentralUri`.

https://github.com/ScaleLeap/amazon-marketplaces/blob/72bd0b4dff82d628b78670626563fbe19cba6128/src/amazon-marketplace.ts#L223-L225

Be careful, there are some unexpected URLs, that don't follow a general pattern:

screenshot-20210716T095000-XqaAh2hI

### LINKED ISSUE #401: Add `vendorCentralAuthUri`
Similar to `sellerCentralAuthUri`, we need `vendorCentralAuthUri`.

The URL follows the same pattern, just the domain changes (see #400).

https://github.com/ScaleLeap/amazon-marketplaces/blob/72bd0b4dff82d628b78670626563fbe19cba6128/src/amazon-marketplace.ts#L256-L258

### PR DESCRIPTION
**Pull request by nguyentoanit**
- closes #366
- closes #399 
- closes #400 
- closes #401

### CODE REVIEW

File: test/__snapshots__/marketplaces.test.ts.snap [hunk 1]
```diff
@@ -227,7 +237,18 @@ AmazonMarketplace {
   "currency": "EGP",
   "id": "ARBP9OOSHTCHU",
   "name": "Egypt",
+  "sellingPartner": Object {
```
Comment (moltar) [inline test/__snapshots__/marketplaces.test.ts.snap:240]: Do we know this for sure, that Egypt is part of `eu-west-1`?

File: src/marketplaces/ae.ts [hunk 2]
```diff
@@ -34,5 +39,6 @@ export const AE = new AmazonMarketplace({
   sellingPartner: {
     region: sellingPartnerRegions.EU,
     sellerCentralAuthUri: sellerCentralAuthUris.NA,
+    vendorCentralAuthUri: vendorCentralAuthUriTemporary, // This will be generated from Vendor Central URI
```
Comment (moltar) [inline src/marketplaces/ae.ts:42]: Not a big fan of the `vendorCentralAuthUriTemporary` approach. I understand the constraint, but we should figure out a better way. We can do this as a separate PR though.

File: test/__snapshots__/marketplaces.test.ts.snap (hunk 1 above)
Comment (nguyentoanit) [reply test/__snapshots__/marketplaces.test.ts.snap:240]: I found them in "[Selling Partner API endpoints](https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#selling-partner-api-endpoints)"

File: src/marketplaces/ae.ts (hunk 2 above)
Comment (nguyentoanit) [reply src/marketplaces/ae.ts:42]: I opened a new issue for this: https://github.com/ScaleLeap/amazon-marketplaces/issues/403
Will think about them. Thanks.
